ikarus/finiteelements/feresulttypes.hh

```diff
@@ -0,0 +1,156 @@
+                           Ikarus::Util::MatricizeGeneric)
+
+  ///=============================
+  /// Registered ResultTypes
+
+  REGISTER_SYMMETRIC_RESULTTYPE(linearStress, worldDim, worldDim);
+  REGISTER_SYMMETRIC_RESULTTYPE(PK2Stress, worldDim, worldDim);
+  REGISTER_SYMMETRIC_RESULTTYPE(cauchyStress, worldDim, worldDim);
+
+  REGISTER_RESULTTYPE(director, worldDim, 1);
+  REGISTER_RESULTTYPE(magnetization, worldDim, 1);
+  REGISTER_RESULTTYPE(gradientNormOfMagnetization, 1, 1);
+  REGISTER_RESULTTYPE(vectorPotential, worldDim, 1);
+  REGISTER_RESULTTYPE(divergenceOfVectorPotential, 1, 1);
+
+  REGISTER_RESULTTYPE(BField, worldDim, 1);
+  REGISTER_RESULTTYPE(HField, worldDim, 1);
+
+  REGISTER_RESULTTYPE(customType, Eigen::Dynamic, Eigen::Dynamic);
+} // namespace ResultType
+
+template <template <typename, int, int> class RT1, template <typename, int, int> class RT2>
+constexpr bool isSameResultType = std::is_same_v<RT1<double, 1, 1>, RT2<double, 1, 1>>;
+
+/**
+ * \brief Container that is used for FE Results. It gives access to the stored value, but can also be used to access
+ * the result in Matrix and Vector form
+ * \tparam RT A specified ResultType
+ * \tparam inputIsVec boolean indicating wether the stored result is in its vector form, defaults to true
+ */
+template <typename RT, bool inputIsVec = true>
+struct ResultTypeContainer : RT
+{
+private:
+  using RegisteredType = typename RT::type;
+  using Traits         = typename RegisteredType::CompileTimeTraits;
+  using ScalarType     = typename RegisteredType::Scalar;
+  static constexpr bool registeredTypeIsDynamic =
+      Traits::RowsAtCompileTime == Eigen::Dynamic or Traits::ColsAtCompileTime == Eigen::Dynamic;
+  static constexpr bool registeredTypeIsVector = Traits::IsVectorAtCompileTime;
+
+public:
+  using VecType    = std::conditional_t<registeredTypeIsDynamic, Eigen::MatrixX<ScalarType>,
+                                     std::invoke_result_t<typename RT::Vectorizer, RegisteredType>>;
+  using MatType    = std::conditional_t<registeredTypeIsDynamic, Eigen::MatrixX<ScalarType>,
+                                     std::invoke_result_t<typename RT::Matricizer, VecType>>;
+  using InputType  = std::conditional_t<inputIsVec, VecType, RegisteredType>;
+  using ResultType = RT;
+
+  const InputType& operator()() const { return value_; }
+
+  /**
+   * \brief Returns the stored value as Vector
+   */
+  auto asVec() const {
+    if constexpr (inputIsVec)
+      return value_;
+    else
+      return typename RT::Vectorizer{}(value_);
+  }
+
+  /**
+   * \brief Returns the stored value as Matrix (if possible)
+   */
+  auto asMat() const {
+    if constexpr (inputIsVec and not registeredTypeIsVector)
+      return typename RT::Matricizer{}(value_);
+    else
+      return value_;
+  }
+
+  explicit ResultTypeContainer() = default;
+
+  void emplace(const InputType&& value) { value_ = std::move(value); }
```

Comment:
const forwarding reference does also not make sense this has to be mutable. Anyway I would remove this emplace function and just create this via the appropriate constructors for ResultTypeContainer. And maybe think about renaming to ResultContainer or ResultWrapper.